### DISCUSSION TITLE
YOLOv8 Instance Segmentation Augmentations

### DISCUSSION THREAD

**Discussion by ghost** [ultralytics/ultralytics#10012] [Q&A]
Hello community,

im trying to augment cityscapes dataset for instance segmentation purposes using Albumentations library. I was wondering, if i apply random crops,rotations etc. is there an embedded function that calculates the new bounding boxes and masks? Its important to know if they are calculated again because during training the augmented images must have the correct bboxes and masks.(image processing will change the locations of the objects)

Can you give me an example how to do that?

Thank you

**Comment by pderrenger** [comment]
Hi there! 👋

Great question! Yes, when you're using Albumentations with YOLOv8 for instance segmentation, the library can indeed adjust bounding boxes and masks according to the applied transformations. When applying transformations like crops or rotations, you'll want to use the `bbox_params` and `mask_params` in your Albumentations pipeline to ensure the bounding boxes and masks are recalculated correctly.

Here's a concise example on how you can set this up:

```python
import albumentations as A
from albumentations.pytorch.transforms import ToTensorV2

# Define your transformations including bbox and mask params
transform = A.Compose([
    A.RandomCrop(width=450, height=450),
    A.HorizontalFlip(p=0.5),
    A.RandomBrightnessContrast(p=0.2),
], bbox_params=A.BboxParams(format='coco', label_fields=['class_labels']),
   mask_params=A.MaskParams())

# Assuming you have image, bboxes and masks loaded
transformed = transform(image=image, bboxes=bboxes, masks=masks, class_labels=class_labels)

# Transformed image, bboxes & masks
transformed_image = transformed['image']
transformed_bboxes = transformed['bboxes']
transformed_masks = transformed['masks']
```

In this setup:
- `bbox_params` ensure bounding boxes are recalculated after transformations. The `format='coco'` specifies the bounding box format. You adjust it according to your dataset's bbox format.
- `mask_params` automatically recalculates masks.
- You'd pass your image, bounding boxes (`bboxes`), and masks (`masks`) to the transformation, and it'll apply the transformations accordingly.

This way, your training data remains consistent with the augmented images. Hope this helps! 😊
